napari/_qt/qt_viewer.py

```diff
@@ -1160,3 +1232,48 @@ def _create_remote_manager(
     qt_poll.events.poll.connect(monitor.on_poll)
 
     return manager
+
+
+def _get_reader_choice_for_file(
```

Comment:
feels kinda like it could/should be a method on QtReaderDialog?  Is this for ease of testing?
maybe this should move to  `qt_reader_dialog.py`?

Reply:
I didn't make it a method because I wasn't sure how to test it as a method, since it pops up the dialog. I'm testing it using a `MockQtReaderDialog` so it needs the `readerDialog` as a parameter.  I've moved it as a function inside `qt_reader_dialog.py` though, because you're right it doesn't really belong in the viewer anymore (or maybe it never did).